### ISSUE TITLE
[Bug]: iOS UserLocation heading indicator not rotating on device with native render mode

### ISSUE THREAD

**Issue by hernanmateo** [rnmapbox/maps#2696]
### Mapbox Implementation

Mapbox

### Mapbox Version

10.11.1

### Platform

iOS

### `@rnmapbox/maps` version

10.0.0-beta.76

### Standalone component to reproduce

```javascript
import React from 'react';
import MapboxGL from '@rnmapbox/maps';

MapboxGL.setAccessToken('MY_TOKEN');

const IssueTest = () => {
  return (
    <MapboxGL.MapView style={{flex: 1}}>
      <MapboxGL.UserLocation
        showsUserHeadingIndicator={true}
        renderMode="native"
      />
      <MapboxGL.Camera
        bounds={{ne: [0.657778, 42.632222], sw: [0.655778, 42.631222]}}
        zoomLevel={16}
        followZoomLevel={16}
        followUserLocation={true}
      />
    </MapboxGL.MapView>
  );
};

export default IssueTest;
```

### Observed behavior and steps to reproduce

The header indicator is frozen

### Expected behavior

The header indicator should rotate following the device compass rotation

**Comment by mfazekas** [comment]
This one can be used to reproduce the issue in simulator

```jsx
import React, { useEffect } from 'react';
import Mapbox from '@rnmapbox/maps';

const IssueTest = () => {
  useEffect(() => {
    Mapbox.locationManager._simulateHeading(20, 0.3);
  }, []);
  return (
    <Mapbox.MapView style={{ flex: 1 }}>
      <Mapbox.UserLocation
        showsUserHeadingIndicator={true}
        renderMode="native"
      />
      <Mapbox.Camera
        bounds={{ ne: [0.657778, 42.632222], sw: [0.655778, 42.631222] }}
        zoomLevel={16}
        followZoomLevel={16}
        followUserLocation={true}
      />
    </Mapbox.MapView>
  );
};

export default IssueTest;

```

**Comment by mfazekas** [comment]
fixed by: #2724
